src/effects/effectchainslot.cpp

```diff
@@ -40,6 +40,19 @@ EffectChainSlot::EffectChainSlot(const QString& group,
             ConfigKey(m_group, "num_effectslots"));
     m_pControlNumEffectSlots->setReadOnly();
 
+    m_pControlNumPresetsAvailable = std::make_unique<ControlObject>(
+            ConfigKey(m_group, "num_presetsavailable"));
+    m_pControlNumPresetsAvailable->set(m_pChainPresetManager->numPresets());
```

Comment:
```suggestion
    m_pControlNumPresetsAvailable->set(numPresets());
```

Reply:
We mustn't call virtual methods in constructors.